The trace facility asks the kernel for the calling thread's identifier
through a list of platform arms; this system has gettid() in its libc.

Copyright (C) 2026 The LikeOS Project

--- libva-2.22.0/va/va_trace.c
+++ libva-2.22.0/va/va_trace.c
@@ -62,6 +62,8 @@
 
 #if defined(__linux__)
 #include <sys/syscall.h>
+#elif defined(__LIKEOS__)
+#include <sched.h>
 #elif defined(__DragonFly__) || defined(__FreeBSD__)
 #include <pthread_np.h>
 #elif defined(__NetBSD__)
@@ -77,6 +79,8 @@
     return GetCurrentThreadId();
 #elif defined(__linux__)
     return syscall(__NR_gettid);
+#elif defined(__LIKEOS__)
+    return gettid();
 #elif defined(__DragonFly__) || defined(__FreeBSD__)
     return pthread_getthreadid_np();
 #elif defined(__NetBSD__)
